### DISCUSSION TITLE
Since VLC is opensource, could it provide handshaking to Slinger for a more robust connection?

### DISCUSSION THREAD

**Discussion by robosmith** [GerryDazoo/Slinger#496] [Ideas]
@GerryDazoo I am willing to look into development enhancements to VLC if that is possible and you're interested in helping me improve Slinger.

**Comment by GerryDazoo** [comment]
Could you define a more robust connection? Down in the guts of Slinger once the stream is started it's basically a read a packet/send a packet loop. So I don't see where "handshaking" with VLC could make a difference.

**Reply by robosmith** [reply]
@GerryDazoo One possibility might be skipping frames if there is a delay in receiving frames sufficient to cause a dropped stream on average. Or maybe a reduction in resolution. Another approach might involve VLC auto initiating a new video stream if it receives confirmation from Slinger that the stream has not been terminated.

**Reply by GerryDazoo** [reply]
I not prepared to bite off on writing the code (I wouldn't even know where to start) and take the overhead to parse the incoming media stream to skip video frames.
 
Why don't you just select a resolution that works in the first place?

VLC can already do that if you configure it properly. 

Dynamically changing the resolution is problematic  because most Media players can't handle the change in the media stream properly. Changing the bit rate might work but then you'd end up with a sub-par picture.  

I've written a minimal Video player for windows that is customized for streaming from Slinger. It will be published in the next release. I'm looking into making an Android version of that player.


Here's what's coming:
[V4.0_release_notes.pdf](https://github.com/GerryDazoo/Slinger/files/10434720/V4.0_release_notes.pdf)

**Reply by robosmith** [reply]
@GerryDazoo I have experimented with using lower resolution (640x480) but it doesn't seem to help to keep the stream alive.
What I was thinking of is an automatic analysis of what resolution or frame rate the link could support without dropping the stream.
